x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/OrdinalGroup.java

```diff
@@ -32,15 +31,10 @@
 
     private Ordinal start, stop;
 
-    protected OrdinalGroup(SequenceKey key, Function<E, Ordinal> extractor) {
-        this.key = key;
+    protected OrdinalGroup(Function<E, Ordinal> extractor) {
         this.extractor = extractor;
     }
 
-    SequenceKey key() {
-        return key;
-    }
-
```

Comment:
The key was removed since all `OrdinalGroups` are now associated with one inside the tracking map. This reduces memory consumption due to keys using different strings that have the same content. When dealing with a large number of keys, this results in a lot of string repetitions.